aws/session/credentials.go

```diff
@@ -187,7 +187,8 @@ func resolveSSOCredentials(cfg *aws.Config, sharedCfg sharedConfig, handlers req
 		oidcClient := ssooidc.New(mySession, cfgCopy)
 		tokenProvider := ssocreds.NewSSOTokenProvider(oidcClient, cachedPath)
 		optFns = append(optFns, func(p *ssocreds.Provider) {
-			p.SSOTokenProvider = tokenProvider
+			p.HasTokenProvider = true
+			p.TokenProvider = *tokenProvider
```

Comment:
Why are we dereferencing here? This doesn't look right. The methods should probably be pointer receivers (looks like they are value receivers currently).